### DISCUSSION TITLE
device types reporting "type" in Homekit

### DISCUSSION THREAD

**Discussion by matt-avila** [dkerr64/homebridge-yolink#88] [Q&A] [answered]
Looking at how the device type is actually reported to HomeKit, specifically water sensors. I have TWO WLS1 (Water Leak Sensor 1) that report as "alarm" sensors... although from what I can see, no internal alarm. Also have TWO YS7904-UC (Water Leak Sensor 3 : PROBE type) sensors that DO seem to report as a LEAK sensor. (really looking to see IF this can be changed by any field reporting to HomeKit).

Im planning adding a relay to the two WLS1 as a trigger for an older valve controller for washing machine shutoff and a "Smart Plug With Power Monitoring" outlet controller to shutoff a washer should either of the WLS1's trigger, essentially the relay will trigger the shutoff valves AND trigger the outlet to shutoff (this "should" close the water valves to keep the water supply from flooding, and turn OFF the washer should the leak be from the machine itself , say a backed-up drain that the washer drain pump is causing the flooding).

While Im not certain of the HomeKit nuances, I do see in the HomeKit App that the leak sensors report differently (although, there appears to be temperature for all 4 leak sensors reporting fine. (another later thought is to be able to trigger on a lek sensors temperature over a longer period of time, mainly for an attic mounted leak sensor in an HVAC pan.... if the temp is below freezing to potentially trigger a heating element).

Lastly, would be great to combine several YoLink devices into a "new" user defined device type - for example water sensor, relay, AND outlet switch into ONE user defined type, and trigger on those variables.

**Comment by dkerr64** [comment, marked as answer]
The WLS1 reports as model YS7903-UC, I have several of them and yes in Apple Home they show as "alarms" for me too.  But if you open up the settings for the device they show as leak sensors.  I do not know why Apple considers them "alarms" in the home summary.

I do not have any YS7904-UC devices, but I do not handle them any differently, they should appear in Apple Home exactly the same.

In theory I could allow a leakSensor to be registered with HomeKit as some other type of device, e.g. a contact sensor.  But I am not keen on doing that when HomeKit supports the native device type, in this case leak sensors.  I do allow it for YoLink power failure alarm (as an outlet or contact sensor) but that is only because HomeKit has no native support for a power failure type.

Bundling several service types under a single accessory is supported in HomeKit.  For example, I include a battery service and temperature service for several device types.  So a leak sensor can report device temperature and battery level and those are hidden within the Apple HomeKit leak sensor settings.  I'm not sure what value there is in allowing any arbitrary combination of devices to be combined together.  I suspect that Apple HomeKit will not handle it well as there is no way to specify what the "outer" device is... apple seems to just pick the most logical.

Finally, as I note in the readme, I would not rely upon Apple HomeKit for automations like turning off a water main.  There are too many things that could go wrong... YoLink's UAC API is not robust enough and it requires internet access.  I strongly encourage you to stick with YoLink's native app and especially device-to-device notifications for something as critical as turning off a water main valve.  That is how I have mine setup.

**Reply by matt-avila** [reply]
Make sense and I appreciate your reply. The homekit reporting seems a little odd, however its picking up something albeit I haven't a clue - especially if your reporting the devices in the same manner. I wasn't really looking to go 'too outside' any boundaries, rather get to some of the subtleties (i,.e. the alarm vs leak reporting)... although its working as it should, so I'm good.

I'm not as up on Apple / Homekit (and especially coding... its been decades) parameters as Id like to be, but in short Im working with several different technologies and Homekit does what its supposed to - puts a nice front end on all the kit in this place. I'm pulling together Lutron Radio RA, hue, YoLink as well as every other IP enabled gadget in the house. from the various HomeBridge code that I've looked at, yours appears the be the best documented and its easy to follow (for a non coder), really appreciate what you've offered!

As for relying on HK or some other automation to shut-off water, Im 100% in agreement that YoLinks D2D communication is the way to go. There are some things I still like wires for, this would be one of them! Thanks again.
